### PR TITLE
Add support back in for GEO IDs

### LINKED ISSUE #104: Add GEO support
GEO support was stripped out in v1.7. See [this PR](https://github.com/nf-core/fetchngs/pull/102) for more information. We should add it back in when we have a robust solution.

### PR DESCRIPTION
**Pull request by ejseqera**
Closes #104

Adding support for GEO IDs back in by:

- Resolving `GSM*` (GEO sample accessions) and `GSE*` (GEO series/experiment IDs which contain multiple GSM IDs) IDs differently 
- Updating the API endpoint for GSM IDs (to use `esearch`) and resolve to SRA IDs directly
-  Use `esummary` to resolve GSE IDs -> GDS UIDs (with `db: gds`) then `esearch` to resolve GDS UIDs -> GSM UIDs -> SRA 
- Updated relevant documentation and removed error handling added in v1.7 to catch GEO IDs

Has been tested on the following GEO IDs:
`GSM4907283`
`GSM465244`
`GSM465247`
`GSE215376`
`GSE18729`

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/fetchngs/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/fetchngs _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by ejseqera** [comment]
We'll need to update the [test-datasets for fetchngs](https://github.com/nf-core/test-datasets/tree/fetchngs) to include a GSM ID (maybe GSE ID aswell?) to small dataset

File: bin/sra_ids_to_runinfo.py [hunk 1]
```diff
@@ -188,9 +188,9 @@ def is_valid(cls, identifier):
 class DatabaseResolver:
     """Define a service class for resolving various identifiers to experiments."""
 
-    _GEO_PREFIXES = {"GSE", "GSM"}
+    _GEO_GSM_PREFIXES = {"GSM"}
+    _GEO_GSE_PREFIXES = {"GDS", "GSE"}
     _SRA_PREFIXES = {
-        "PRJNA",
```
Comment (drpatelh) [inline bin/sra_ids_to_runinfo.py:193]: Why was this line removed?

**Comment by drpatelh** [comment]
This is the GEO id I was using previously for testing [`GSM4907283`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4907283) before we removed support (117480 raw reads)

**Comment by ejseqera** [comment]
Made a PR to add [GSE214215 ](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE214215) for a series/experiment that is composed of 2 SRA runs, see https://github.com/nf-core/test-datasets/pull/887.

File: bin/sra_ids_to_runinfo.py (hunk 1 above)
Comment (drpatelh) [reply bin/sra_ids_to_runinfo.py:193]: Cool. Let's add a separate issue for this so we can track and add in future releases.

**Review by drpatelh — APPROVED** [review]
Gorgeous!
